src/templates/blog-post.tsx

```diff
@@ -46,7 +46,10 @@ export const Head = ({
   const { author, seoMetaText, title, publicationDate, heroImage } =
     data.contentfulBlogPost;
 
-  const shareImagePath = heroImage.shareImage.resize.src;
+  let shareImagePath = '';
+  if (heroImage && heroImage.shareImage && heroImage.shareImage.resize) {
```

Comment:
FYI, the equation could also be written in this short form:
```ts
  if (heroImage?.shareImage?.resize) {
```

Or the complete `if` block replaced by:
```ts
const shareImagePath = heroImage?.shareImage?.resize?.src || '';
```